### ISSUE TITLE
Custom reduction operator documentation

### ISSUE THREAD

**Issue by ian-bertolacci** [chapel-lang/chapel#8139]
I was curious about the state of custom reductions in Chapel, and found the [ChapelReduce.chpl](https://github.com/chapel-lang/chapel/blob/master/modules/internal/ChapelReduce.chpl) source.

I think that it would be relatively simple to write some documentation for this, and would be super useful for converting more functional language developers (python comes to mind) to Chapel.

**Comment by bradcray** [comment]
I agree that this is a gap in our current documentation.  Part of the reason we've dragged our feet on it is that we expect the reduction interface to change, particularly as we enable support for partial reductions (reducing only across rows or cols of a 2D array, e.g.), and so haven't wanted to people to commit too much code to the current interface before we have more faith in it.  That said, it may very well be that we've erred too much on the side of caution...

Probably the right way to do this at present would be to add either a [technical note](https://chapel-lang.org/docs/latest/technotes/index.html) (where we tend to document things that aren't ready for the spec / users guide yet) or a [primer](https://chapel-lang.org/docs/latest/primers/index.html).

**Comment by mppf** [comment]
@vasslitvinov @bradcray - I wonder if we could create even just a simple primer creating a custom reduction as an example, for this release?

**Comment by vasslitvinov** [comment]
Brad's earlier argument still holds - we are on the brink (TM) of changing the reduction interface. Although I am open to creating a primer or a tech note. Preferably, a tech note, because it can be less committal and I can do it after code freeze.

There is a sample reduction op class in the [tech note on reduce intents](https://chapel-lang.org/docs/technotes/reduceIntents.html), which btw I need to update.

**Comment by bradcray** [comment]
I agree that if we did primer or tech note, tech note seems more appropriate given the amount of flux we anticipate this continuing to have.

I wouldn't lose sleep if this didn't happen for this release given that it seems a shame to write it and have it change almost immediately (he said, fingers crossed).
